### ISSUE TITLE
Thick border around the terminal with padding and margin width configuration

### ISSUE THREAD

**Issue by ciscohack** [kovidgoyal/kitty#4863]
Hello Expert,

I need your assistance on this issue . I am using  windows margin and padding setting in my kitty config and observed thick border around the terminal when select theme from kitty +kitten themes. to reproduce the issue put below configuration in kitty.conf and select the theme from inbuilt  method. you will see ugly looking terminal.

window_margin_width 7
window_padding_width 7

hide_window_decorations yes
macos_show_window_title_in none
macos_titlebar_color background
resize_in_steps yes

I have never see this issue with iterm2 or any other terminal almost all have padding option and it terminal works consistently with this setting

<img width="344" alt="image" src="https://user-images.githubusercontent.com/11479557/159581144-105df72e-0821-44a8-9780-5e6f26f41dbd.png">

<img width="388" alt="image" src="https://user-images.githubusercontent.com/11479557/159581301-1f20a11e-37f6-4762-9b9a-453e3d90a802.png">

**Comment by kovidgoyal** [comment]
As is noted in the docs for macos_titlebar_color, it's a hack. If you
use it, you are going to get some non-optimal behavior. In this case
once you have changed the colors, restarting kitty will fix the titlebar
color.

**Comment by kovidgoyal** [comment]
Actually I tested those settings on my macOS machine and they work fine. The titlebar color is updated when the kitten selects a new theme, as expected.

**Comment by page-down** [comment]
Do you think this deserves a fix?

```shell
kitty --config=NONE -o window_margin_width=8 -o window_padding_width=8 -o background=blue
```

- `kitty +kitten themes`
- Switch to Light
- ESC
- The padding area remains white.

![](https://user-images.githubusercontent.com/57713011/159612931-bc9355e7-c52a-45c2-baf1-579fbedd0c8a.png)

Also I found that the background color of the margin area is not consistent when in single or multiple windows.
However I believe this works as expected.

- single window, margin bg = window bg
- multi windows, margin bg = default bg

**Comment by kovidgoyal** [comment]
Yes that's by design. the kitten only changes colors of the current window margin colors come from overall defaults, unless there is only a single window. As for resetting the padding color after the kitten is aborted, sure, that should be fixed.

**Comment by ciscohack** [comment]
> As is noted in the docs for macos_titlebar_color, it's a hack. If you use it, you are going to get some non-optimal behavior. In this case once you have changed the colors, restarting kitty will fix the titlebar color.

@kovidgoyal Thanks for testing. strange for me that is not working consistently and annoying rest ..i never used terminal like kitty. 

@kovidgoyal  Sorry but if you don't mind. May i have your kitty.conf copy to try on my macOS.. it must be configured optimal way

**Comment by kovidgoyal** [comment]
I used your kitty.conf to test

**Comment by ciscohack** [comment]
@kovidgoyal Ah okay with just those 4 lines. okay .. Anyway i request you in general config file 😊 master chef cooked kitty.conf file will all perfect options in place .. Thanks once again

**Comment by kovidgoyal** [comment]
I just use the defaults, apart from fonts. After all the defaults are
what suits me.

**Comment by ciscohack** [comment]
Thanks for clarification. only glitch that border padding different color is annoying me rest all is great .. will see how to overcome. in macos you able to move kitty terminal around the screen that is one thing i am not able to do after this seeting 

hide_window_decorations yes
macos_show_window_title_in none
